configure.ac

```diff
@@ -41,17 +41,17 @@ AC_ARG_ENABLE([debug],
 	      [AS_HELP_STRING([--enable-debug],
 			      [Enable debugging @<:@default=no@:>@])
 	      ],
-	      [CFLAGS="$CFLAGS -g -O0 -Wall -Wextra -Wundef -Wno-unused-parameter -Wno-sign-compare -Wno-missing-field-initializers"
+	      [CFLAGS="-g -O0 -Wall -Wextra -Wundef -Wno-unused-parameter -Wno-sign-compare -Wno-missing-field-initializers $CFLAGS"
 	       dbg=1],
 	      [enable_debug=no
                dbg=0])
 
 AC_DEFINE_UNQUOTED([ENABLE_DEBUG],[$dbg],
                    [defined to 1 if libfabric was configured with --enable-debug, 0 otherwise])
 
-dnl Fix autoconf's habit of adding -g -O2 by default
-AS_IF([test -z "$CFLAGS"],
-      [CFLAGS='-fvisibility=hidden -O2 -DNDEBUG -Wall -Wundef'])
+# Fix autoconf's habit of setting CFLAGS="-g -O2" by default while still
+# allowing the user to explicitly set CFLAGS=""
+: ${CFLAGS="-fvisibility=hidden -O2 -DNDEBUG -Wall -Wundef"}
```

Comment:
@bturrubiates points to http://www.tldp.org/LDP/abs/html/parameter-substitution.html.  Ok.